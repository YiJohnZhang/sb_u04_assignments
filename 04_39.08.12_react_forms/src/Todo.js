import React from 'react';

function Todo({todoText, completed, deleteTodoEntry, editTodoEntry, toggleTodoEntryStatus}){

	return (
		<li>
			<button data-test-id="deleteTodo" onClick={deleteTodoEntry}>x</button>&nbsp;&nbsp;&nbsp;
			{/* <span style={completed === true ? {textDecoration:line-through} : ''} onClick={toggleTodoStatus}>{todoText}</span> */}
			<span
				style={completed === true ? {textDecorationLine: 'line-through', textDecorationStyle: 'solid', fontStyle:'italic'} : {textDecoration:'none', fontStyle:'normal'}}
				onClick={toggleTodoEntryStatus}>
					{todoText}
			</span>

		</li>
	);

}

export default Todo;