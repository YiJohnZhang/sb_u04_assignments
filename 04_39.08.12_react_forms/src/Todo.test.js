import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import Todo from './Todo';
// import TodoList from './TodoList';

test('smoke test', () => {

	render(<Todo />);

});

test('snapshot', () => {
	
	const {asFragment} = render(<Todo />);
	expect(asFragment()).toMatchSnapshot();

});

test('snapshot 2: some content with `asdf`', () => {
	
	const {asFragment} = render(<Todo todoText='asfd'/>);
	expect(asFragment()).toMatchSnapshot();

});

// point is: i can't test it, i need to test the parent component for the delete behavior
// test('fireEvent: text change, submit & remove todo entry', () => {

// 	console.log(TodoList);

// 	const { queryByText, queryByRole } = render(<Todo todoText='asfd' />);
// 	// console.log(queryByText('x'));
// 	// const deleteToDoEntryButton = queryByRole('button', {'description':'x'});
// 	const deleteToDoEntryButton = queryByText('x');

// 	expect(queryByText('asfd')).toBeInTheDocument();
// 	fireEvent.click(deleteToDoEntryButton);
// 	// expect(queryByText('asfd')).not.toBeInTheDocument();
	

// });