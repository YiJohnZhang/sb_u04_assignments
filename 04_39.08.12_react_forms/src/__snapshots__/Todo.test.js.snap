// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <li>
    <button
      data-testid="deleteTodo"
    >
      x
    </button>
       
    <span
      style="text-decoration: none; font-style: normal;"
    />
  </li>
</DocumentFragment>
`;

exports[`snapshot 2: some content with \`asdf\` 1`] = `
<DocumentFragment>
  <li>
    <button
      data-testid="deleteTodo"
    >
      x
    </button>
       
    <span
      style="text-decoration: none; font-style: normal;"
    >
      asfd
    </span>
  </li>
</DocumentFragment>
`;
